includes/class-wc-dintero-hp-checkout.php

```diff
@@ -2040,20 +2070,77 @@ private function get_iframe(){
 		}
 
 		if (WC()->shipping->get_packages() && WC()->session->get( 'chosen_shipping_methods' )[0]) {
-			$dintero_shipping_options = array(
-				0 => array(
-					'id' => (string)$selectedShippingReference['id'],
-					'line_id' => 'shipping_method',
-					'amount' => (int)$this->get_shipping_amount(),
-					'vat_amount' => $this->get_shipping_tax_amount(),
-					'vat' => $this->get_shipping_tax_rate(),
-					'title' => 'Shipping: ' . $this->get_shipping_name(),
-					'description' => '',
-					'delivery_method' => 'delivery',
-					'operator' => '',
-					'operator_product_id' => (string)$selectedShippingReference['instance_id'],
-				)
-			);
+			$dintero_shipping_options = array();
+			$isShippingInIframe = 'yes' == WCDHP()->setting()->get('shipping_method_in_iframe');
+
+			if($isShippingInIframe){
+				$packages         = WC()->shipping->get_packages();
+				$tax_display      = get_option( 'woocommerce_tax_display_cart' );
+				$j = 0;
+				foreach ( $packages as $i => $package ) {
+					$chosen_method = isset( WC()->session->chosen_shipping_methods[ $i ] ) ? WC()->session->chosen_shipping_methods[ $i ] : '';
+					foreach ( $package['rates'] as $method ) {
+						$method_id   = $method->id;
+						$method_name = $method->label;
+
+						if ( $separate_sales_tax || 'excl' === $tax_display ) {
+							$method_price = intval( round( $method->cost, 2 ) * 100 );
+						} else {
+							$method_price = intval( round( $method->cost + array_sum( $method->taxes ), 2 ) * 100 );
+						}
+
+						if ( array_sum( $method->taxes ) > 0 && ( ! $separate_sales_tax && 'excl' !== $tax_display ) ) {
+							$method_tax_amount = intval( round( array_sum( $method->taxes ), 2 ) * 100 );
+							$method_tax_rate   = intval( round( ( array_sum( $method->taxes ) / $method->cost ) * 100, 2 ) * 100 );
+						} else {
+							$method_tax_amount = 0;
+							$method_tax_rate   = $this->get_shipping_tax_rate();
+						}
+						$method_selected    = $method->id === $chosen_method ? true : false;
+						$dintero_shipping_options[] = array(
+							'id'          => $method_id,
+							'line_id' 	  => 'shipping_method_'.$j,
+							'title'       =>  $method_name,
+							'amount'      =>  (int) $method_price,
+							'vat_amount'  =>(int) $method_tax_amount,
+							'vat'    => $method_tax_rate,
+							'description' => '',
+							'delivery_method' => 'delivery',
```

Comment:
delivery_method might vary:

The built in `Local pickup`-options should be `pick_up`